pkg/generate/app/cmd/newapp.go

```diff
@@ -500,7 +500,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if ref.Input().Uses.IsDockerBuild() {
 					exposed, ok := ref.Input().Uses.Info().Dockerfile.GetDirective("EXPOSE")
 					if ok {
-						if input.Info == nil {
+						if !(input.Info != nil && input.Info.Config != nil) {
```

Comment:
I don't think this is the right fix.  if input.Info is not nil but input.Info.Config is nil, you're going to clobber the input.Info object.

(also that if check could be written more simply as "if input.Info==nil || input.Info.Config==nil", but that's moot).

The right fix here is to fix the imageref.go FromName method to always populate the Config field, and/or to just populate the Config field with an empty value if the Config field is nil in the existing Info object.